EmployeeManagementSystem/Program.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using CommandLine;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace EmployeeManagement;
+
+class Program
+{   
+    private readonly IBAL _bal;
+    private readonly ILogger _console;
+    private readonly string filePath = "";
+    public Program(IBAL bal, ILogger console)
+    {
+        _bal = bal;
+        _console = console;
+    }
+    
+    public static void Main(string[] args)
+    {
+        Parser.Default.ParseArguments<Options>(args)
+        .WithParsed(options =>
+        {
+            ILogger console = new ConsoleLogger();
+            IBAL bal = new EmployeeBAL(new EmployeeDAL(console), console);
+            Program program = new Program(bal, console);
+            if(options.Add)
+            {
+                program._bal.Add(program.EmployeeDataInput());
+            }
+            else if(options.Filter)
+            {
+                program.FilterAndDisplay();
+            }
+            else if(options.Edit != 0)
+            {
+                program._bal.Update(program.Update());
+            }
+            else if (options.Delete != null && options.Delete.Any())
+            {
+                program._bal.Delete(options.Delete);
+            }
+            else if(options.Display)
+            {
+                List<Employee> employees = program._bal.GetAllEmployees();
+                program.Display(employees);
+            }
+            else
+            {
+                program._console.LogError("Invalid Command");
+            }
+        });
+    }
+
+    public Employee EmployeeDataInput()
+    {
+        _console.LogInfo("Employee Number: ");
```

Comment:
Don't use logger for printing messages to console. The purpose of logger is to log messages. If you want to display messages to user, create separate functions for it. Something like DisplayMessage